docs/dev.rst

```diff
@@ -1,20 +1,570 @@
+
+
+GitHub Repository Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+GitHub repository dependencies create a dynamic dependency between the current project and another project on GitHub that uses CumulusCI to manage its dependencies.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+ 
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+
+When ``update_dependencies`` runs, these steps are taken against the referenced repository.
+
+* Look for the ``cumulusci.yml`` file and parse if found.
+* Determine if the project has subfolders under unpackaged/pre.  If found, deploys them first.
+* Determine if the project specifies any dependencies in the ``cumulusci.yml`` file.  If found, deploys them next in the queue.
+* Determine if the project has a namespace configured in the ``cumulusci.yml`` file. If found, treats the project as a managed package unless the unmanaged option is also ``True``.
+* If the project has a namespace and is not set for unmanaged, use the GitHub API to get the latest release and install it.
+* If the project is an unmanaged dependency, the ``src`` or ``force-app`` directory is deployed.
+* Determine if the project has subfolders under unpackaged/post. If found, deploys them next. Namespace tokens are replaced with ``namespace__`` or an empty string, depending on if the dependency is considered managed or unmanaged.
+
+
+
+Reference Unmanaged Projects
+****************************
+
+If the referenced repository does not have a namespace configured, or if the dependency specifies the ``unmanaged`` option as ``True``, the repository is treated as an unmanaged repository.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+              unmanaged: True
+
+..
+
+    The EDA repository is configured for a namespace, but the dependency  specifies ``unmanaged: True``, so the dependency deploys unmanaged EDA and its dependencies.
+
+
+
+Reference a Specific Tag
+************************
+
+To reference a version other than HEAD and the latest production release, use the ``tag`` option to specify a tag from the target repository. This option is most useful for testing against beta versions of underlying packages, or recreating specific org environments for debugging.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+              tag: beta/1.47-Beta_2
+
+..
+
+    The EDA repository's tag, ``beta/1.47-Beta_2``, is used instead of the latest production release of EDA (1.46, for this example). This tag lets a build environment use features in the next production release of EDA that are already merged but not yet included in a production release.
```

Comment:
We could also note the `include_beta` flag here, to avoid pinning a specific version, which is a fairly rare use case.

Reply:
@Br4nd0R @davidmreed: Also, isn’t the example already outdated -- or will be -- when we talk about latest production releases ("1.46, for this example")?

Reply:
This shouldn't matter.